assets/js/wfparameters/RenameEntries.test.js

```diff
@@ -0,0 +1,194 @@
+        api = {
+            inputColumns: jsonResponseMock(columns),
+            onParamChanged: jest.fn().mockReturnValue(Promise.resolve()),
+            deleteModule: jest.fn().mockReturnValue(Promise.resolve())
+        };
+        mockAPI(api);
+    });
+
+    it('Adds all columns to entries and turns off loadAll when loadAll is set to true', (done) => {
+        // This test corresponds to behavior when added from module library.
+        var changeLoadAllMock = jest.fn().mockReturnValue(Promise.resolve());
+
+        let tree = mount(<RenameEntries
+            loadAll={true}
+            changeLoadAll={changeLoadAllMock}
+            entries={JSON.stringify({})}
+            wfModuleId={WFM_ID}
+            paramId={PARAM_ID}
+        />);
+
+        setImmediate(() => {
+            // Got the tip to call .update() in this thread:
+            // https://github.com/airbnb/enzyme/issues/1233#issuecomment-343449560
+            tree.update();
+
+            // Turns off loadAll after initial load
+            expect(changeLoadAllMock.mock.calls).toHaveLength(1);
+            expect(changeLoadAllMock.mock.calls[0][0]).toBe(false);
+
+            // Updates entries param to include every column, "renamed" to their original name
+            expect(api.onParamChanged.mock.calls).toHaveLength(1);
+            expect(api.onParamChanged.mock.calls[0][0]).toBe(PARAM_ID);
+            let changedParam = JSON.parse(api.onParamChanged.mock.calls[0][1].value);
+            expect(changedParam['name']).toBe('name');
+            expect(changedParam['build_year']).toBe('build_year');
+            expect(changedParam['narrative']).toBe('narrative');
+            expect(changedParam['cornerstone']).toBe('cornerstone');
+            tree.unmount();
+            done();
+        });
+    });
+
+    it('Displays all columns in entries after loadAll is set to false', (done) => {
+        let tree = mount(<RenameEntries
+            loadAll={false}
+            changeLoadAll={jest.fn()}
+            entries={JSON.stringify(testEntries)}
+            wfModuleId={1}
+            paramId={2}
+        />);
+
+        setImmediate(() => {
+            // Got the tip to call .update() in this thread:
+            // https://github.com/airbnb/enzyme/issues/1233#issuecomment-343449560
+            tree.update();
+            expect(tree.find('.rename-input')).toHaveLength(2);
+            expect(tree.find('.rename-input').get(0).props.value).toEqual('host_name');
+            expect(tree.find('.rename-input').get(1).props.value).toEqual('nrtv');
+            tree.unmount();
+            done();
+        });
+    });
+
+    it('Updates parameter upon input completion via blur', (done) => {
+        let tree = mount(<RenameEntries
+            loadAll={false}
+            changeLoadAll={jest.fn()}
+            entries={JSON.stringify(testEntries)}
+            wfModuleId={WFM_ID}
+            paramId={PARAM_ID}
+        />);
+
+        setImmediate(() => {
```

Comment:
Next time, I think we should use async-function tests instead of async-callback tests, since async tests add unnecessary nesting. See `GoogleFileSelect.test.js` for an example.